xblock_skytap/skytap.py

```diff
@@ -129,10 +138,118 @@ def student_view(self, context):
+                )
+            )
+        return boomi_configuration
+
+    def get_boomi_url(self):
+        """
+        Using the "boomi_configuration" in the XBLOCK_SETTINGS, construct the authenticated Boomi endpoint URL.
+        """
+        boomi_configuration = self.get_boomi_configuration()
+
+        auth_string = "{}:{}".format(boomi_configuration['username'], boomi_configuration['token']).encode('ascii')
+        base64_auth_string = base64.b64encode(auth_string)
+
+        return "{base_url};boomi_auth={base64_auth_string}".format(
+            base_url=urljoin(boomi_configuration['base_url'], boomi_configuration['endpoint']),
+            base64_auth_string=base64_auth_string.decode('ascii')
+        )
+
+    @staticmethod
+    def raise_error(message='An unknown error occurred.', exception=False):
+        """
+        Given an error message, log the error and raise a JsonHandlerError to invoke an error response.
+        """
+        if exception:
+            log.exception(message)
+        else:
+            log.error(message)
+
+        raise JsonHandlerError(500, message)
+
     @XBlock.json_handler
     def launch(self, keyboard_layout, suffix=""):
         """
-        Launch Skytap environment.
+        Launch Skytap environment and return the resulting sharing portal URL.
         """
         self.preferred_keyboard_layout = keyboard_layout
-        return {}
+
+        # Gather information from the runtime and settings
+        current_user = self.get_current_user()
+        current_course = self.get_current_course()
+        if current_user is None:
+            self.raise_error('Unable to fetch the current user from the runtime.')
+        if current_course is None:
+            self.raise_error('This block usage is not associated with a course.')
+        current_user_email = current_user.emails.pop()
+        current_course_name = current_course.course
+        current_course_run = current_course.run
+        try:
+            boomi_url = self.get_boomi_url()
+        except (BoomiConfigurationInvalidError, BoomiConfigurationMissingError):
+            self.raise_error('The Skytap XBlock is improperly configured.', exception=True)
+
+        # Fetch the sharing portal URL from Boomi
+        response = requests.post(
+            boomi_url,
+            json={
+                'email': current_user_email,
+                'keyboard_layout': keyboard_layout,
+                'course_name': current_course_name,
+                'course_run': current_course_run,
+            },
+            headers={'Accept': 'application/json'},
+        )
+
+        # Handle response errors
+        try:
+            response_json = response.json()
+        except JSONDecodeError:
+            log.error('The Boomi endpoint returned the following non-JSON response content: %s', response.content)
+            self.raise_error('The Skytap launch service returned a malformed response.', exception=True)
```

Comment:
Ah, took me a second to find [an example](https://github.com/open-craft/xblock-leaderboard/blob/master/leaderboards/leaderboard.py#L34-L36), but I got this added in. Thanks!